be/src/exprs/agg/map_agg.h

```diff
@@ -0,0 +1,151 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <fmt/format.h>
+
+#include "column/binary_column.h"
+#include "column/column.h"
+#include "column/column_helper.h"
+#include "column/fixed_length_column.h"
+#include "column/hash_set.h"
+#include "column/map_column.h"
+#include "column/type_traits.h"
+#include "exprs/agg/aggregate.h"
+#include "exprs/function_context.h"
+#include "gutil/casts.h"
+#include "util/phmap/phmap.h"
+#include "util/time.h"
+
+namespace starrocks {
+
+template <LogicalType KT, typename MyHashMap = std::map<int, Datum>>
+struct MapAggAggregateFunctionState : public AggregateFunctionEmptyState {
+    using KeyColumnType = RunTimeColumnType<KT>;
+    using KeyType = typename SliceHashSet::key_type;
+
+    MyHashMap hash_map;
+    void update(MemPool* mem_pool,
+                const KeyColumnType& arg_key_column,
+                const Column& arg_value_column,
+                size_t offset, size_t count) {
+        for (int i = offset; i < offset + count; i++) {
+            if constexpr (!lt_is_string<KT>) {
+                auto key = arg_key_column.get_data()[i];
+                auto value = arg_value_column.get(i);
+                hash_map.emplace(key, value);
+            } else {
+                auto raw_key = arg_key_column.get_slice(i);
+                KeyType key(raw_key);
+                uint8_t* pos = mem_pool->allocate(key.size);
+                memcpy(pos, key.data, key.size);
+                hash_map.emplace(Slice(pos, key.size), arg_value_column.get(i));
+            }
+        }
+    }
+};
+
+template <LogicalType KT, typename MyHashMap = std::map<int, int>>
+class MapAggAggregateFunction final
+        : public AggregateFunctionBatchHelper<MapAggAggregateFunctionState<KT, MyHashMap>,
+                                              MapAggAggregateFunction<KT, MyHashMap>> {
+public:
+    using KeyColumnType = RunTimeColumnType<KT>;
+    using CppType = RunTimeCppType<KT>;
+
+    void reset(FunctionContext* ctx, const Columns& args, AggDataPtr state) const override {
+        auto& state_impl = this->data(state);
+        if (!state_impl.hash_map.empty()) {
+            state_impl.hash_map.clear();
+        }
+    }
+
+    void update(FunctionContext* ctx, const Column** columns, AggDataPtr __restrict state,
+                size_t row_num) const override {
+        // Key could not be null.
+        if ((columns[0]->is_nullable() && columns[0]->is_null(row_num)) || columns[0]->only_null()) {
+            return;
+        }
+        const auto& key_column = down_cast<const KeyColumnType&>(*ColumnHelper::get_data_column(columns[0]));
+        this->data(state).update(ctx->mem_pool(), key_column, *columns[1], row_num, 1);
```

Comment:
key_column is nullable column.
```
for input { (null, "v1"), "k1": "v2" } our expect result should be [(null, "v1"), "k1": "v2"] rather than [("", "v1"), "k1": "v2"]
```

Reply:
The NULL will be ignored in Trino. So I just want to keep the behaviors the same.

Reply:
but now for input 
```
{ (null, "v1"), "k1": "v2" }
```
you will get result
```
{ ("", "v1"), "k1": "v2" }
```

Reply:
Thanks for the reply!

Reply:
![image](https://github.com/user-attachments/assets/78e3e826-ca61-4f8e-9340-d410bdfd6f4b)
Here is the test about it.